pkcs11/_pkcs11.pyx

```diff
@@ -293,6 +293,18 @@ class SearchIter:
             self.session._operation_lock.release()
 
 
+def _finalize_template(template_, template):
```

Comment:
I like to avoid functions that mutate their inputs. I would do this like:

```python
def merge_templates(default_template, *user_templates):
    template = default_template.copy()

    for user_template in user_templates:
        if user_template is not None:
            template.update(user_template)

    return {
        key: value
        for key, value in template.items()
        if value is not DEFAULT
    }
```

It's also a good opportunity to name things what they are.